mitmproxy/addons/dns_resolver.py

```diff
@@ -19,129 +19,41 @@ def __init__(self, response_code: int) -> None:
         self.response_code = response_code
 
 
-async def resolve_question_by_name(
-    question: dns.Question,
-    loop: asyncio.AbstractEventLoop,
-    family: socket.AddressFamily,
-    ip: Callable[[str], ipaddress.IPv4Address | ipaddress.IPv6Address],
-) -> Iterable[dns.ResourceRecord]:
-    try:
-        addrinfos = await loop.getaddrinfo(
-            host=question.name, port=0, family=family, type=socket.SOCK_STREAM
-        )
-    except socket.gaierror as e:
-        if e.errno == socket.EAI_NONAME:
-            raise ResolveError(dns.response_codes.NXDOMAIN)
-        else:
-            # NOTE might fail on Windows for IPv6 queries:
-            # https://stackoverflow.com/questions/66755681/getaddrinfo-c-on-windows-not-handling-ipv6-correctly-returning-error-code-1
-            raise ResolveError(dns.response_codes.SERVFAIL)  # pragma: no cover
-    return map(
-        lambda addrinfo: dns.ResourceRecord(
-            name=question.name,
-            type=question.type,
-            class_=question.class_,
-            ttl=dns.ResourceRecord.DEFAULT_TTL,
-            data=ip(addrinfo[4][0]).packed,
-        ),
-        addrinfos,
-    )
-
-
-async def resolve_question_by_addr(
-    question: dns.Question,
-    loop: asyncio.AbstractEventLoop,
-    suffix: str,
-    sockaddr: Callable[[list[str]], tuple[str, int] | tuple[str, int, int, int]],
-) -> Iterable[dns.ResourceRecord]:
-    try:
-        addr = sockaddr(question.name[: -len(suffix)].split(".")[::-1])
-    except ValueError:
-        raise ResolveError(dns.response_codes.FORMERR)
-    try:
-        name, _ = await loop.getnameinfo(addr, flags=socket.NI_NAMEREQD)
-    except socket.gaierror as e:
-        raise ResolveError(
-            dns.response_codes.NXDOMAIN
-            if e.errno == socket.EAI_NONAME
-            else dns.response_codes.SERVFAIL
-        )
-    return [
-        dns.ResourceRecord(
-            name=question.name,
-            type=question.type,
-            class_=question.class_,
-            ttl=dns.ResourceRecord.DEFAULT_TTL,
-            data=dns.domain_names.pack(name),
+class DnsResolver:
+    def load(self, loader):
+        loader.add_option(
+            "use_hosts_file",
```

Comment:
We generally suck at this, but maybe we should prefix the new options with `dns_`?